test/src/mac/macPackagerTest.ts

```diff
@@ -16,7 +16,7 @@ test.ifMac.ifAll("two-package", () => assertPack("test-app", {
       electronLanguages: ["bn", "en"]
     },
     //tslint:disable-next-line:no-invalid-template-strings
-    artifactName: "${name}-${version}-${os}.${ext}",
+    artifactName: "${name}-${version}-${os}-${arch}.${ext}",
```

Comment:
From my testing I think this might actually be:

`artifactName: "${name}-${version}-${arch}-${os}.${ext}",`

and I think it's generated in https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/targets/ArchiveTarget.ts#L28

Reply:
Hmm, that is true if an `artifactName` template is not provided in the build config. I assumed this test was also validating `artifactName` as opposed to using the `defaultPattern` you linked in ArchiveTarget.ts.
I updated this test to have the `${arch}` template explicitly specified as otherwise, the dmg/zip will overwrite itself when building for multiple targets.

Reply:
I think I've been seeing something like this. When I'm exporting and uploading DMGs, the -arm64 DMGs don't get published - it's weird. It doesn't seem to happen every time, but I can't figure out what's causing it to happen.